COMPILING.md

```diff
@@ -282,6 +283,92 @@ To build full curses version with localizations and lua enabled:
 
 Make sure that `x86_64-apple-darwin15-clang++` is in `PATH` environment variable.
 
+## Cross-compile to Android from Linux
+
+The Android build uses [Gradle](https://gradle.org/) to compile the java and native C++ code, and is based heavily off SDL's [Android project template](https://hg.libsdl.org/SDL/file/f1084c419f33/android-project). See the official SDL documentation [README-android.md](https://hg.libsdl.org/SDL/file/f1084c419f33/docs/README-android.md) for further information.
+
+The Gradle project lives in the repository under `android/`. You can build it via the command line or open it in [Android Studio](https://developer.android.com/studio/). For simplicity, it only builds the SDL version with all features enabled, including tiles, sound, localization and lua.
+
+### Dependencies
+
+  * Java JDK 8
+  * SDL2 (tested with 2.0.8, though a custom fork is recommended with project-specific bugfixes)
+  * SDL2_ttf (tested with 2.0.14)
+  * SDL2_mixer (tested with 2.0.2)
+  * SDL2_image (tested with 2.0.3)
+  * libintl-lite (tested with a custom fork of libintl-lite 0.5)
+  * lua (tested with lua 5.1.5)
+  
+Some dependencies need their entire source code copied under the Android project folder so they can be built for Android - see below.
+
+### Setup
+
+Install Linux dependencies. For a desktop Ubuntu installation:
+
+    sudo apt-get install lua5.2 openjdk-8-jdk-headless
+
+Install Android SDK and NDK:
+
+    wget https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip
+    unzip sdk-tools-linux-4333796.zip -d ~/android-sdk
+    rm sdk-tools-linux-4333796.zip
+    ~/android-sdk/tools/bin/sdkmanager --update
```

Comment:
```
zhilkin@ubuntu:~$ ~/android-sdk/tools/bin/sdkmanager --update
Warning: File /home/zhilkin/.android/repositories.cfg could not be loaded.      
[=======================================] 100% Computing updates...             
```

Reply:
I believe this warning is "normal" for this Android toolchain version. It appears if the file it's looking for doesn't exist locally, but doesn't impact anything after that: https://askubuntu.com/questions/885658/android-sdk-repositories-cfg-could-not-be-loaded